cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -124,9 +126,48 @@
    * Create a schedule which is triggered whenever at least a certain number of new partitions
    * are added to a certain dataset.
    *
-   * @param datasetName the name of the dataset in the same namespace of the app
+   * @param datasetName   the name of the dataset in the same namespace of the app
    * @param numPartitions the minimum number of new partitions added to the dataset to trigger the schedule
    * @return this {@link ScheduleBuilder}
    */
   ScheduleCreationSpec triggerOnPartitions(String datasetName, int numPartitions);
+
+  /**
+   * Create a schedule which is triggered based upon the specified program in a namespace, application, application
+   * version, and a specific program status.
+   *
+   * @param programNamespace   the namespace where this program is defined
+   * @param application        the name of the application where this program is defined
+   * @param applicationVersion the version of the application
+   * @param programType        the type of the program, as supported by the system
+   * @param program            the name of the program
+   * @param programStatus      the status of the program to trigger the schedule
+   * @return this {@link ScheduleBuilder}
+   */
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, String application, String applicationVersion,
```

Comment:
Do we allow cross namespace trigger? How does it work with security? Generally speaking, if the owner of the schedule don't have permission on the other program, can it be triggered by another program (whether it's in same namespace or not).